### DISCUSSION TITLE
Allow switching between Vercel edge and serverless on a per route/api basis

### DISCUSSION THREAD

**Discussion by andrewdoro** [withastro/roadmap#559] [Proposal]
# Summary

Allow switching between Vercel `edge` and `serverless` on a per route/api basis
``` ts
---
//index.astro
export const runtime = "edge";
...
---

(...)
``` 
  
# Background & Motivation

In the current implementation you can only choose one runtime for all the routes/api's by importing one of the adapters:
-   `edge`: SSR inside an  [Edge function](https://vercel.com/docs/concepts/functions/edge-functions).
-   `serverless`: SSR inside a  [Node.js function](https://vercel.com/docs/concepts/functions/serverless-functions).
-   `static`: generates a static website following Vercel’s output formats, redirects, etc.
``` ts
import { defineConfig } from 'astro/config';
import vercel from '@astrojs/vercel/serverless';

export default defineConfig({
  output: 'server',
  adapter: vercel(),
});
``` 
I've found multiple problems when using only the `edge` runtime because some libraries have dependencies incompatible with the current standards, or don't support edge at all. It would be helpful for these cases to change the mode to `serverless`, while keeping other on the `edge`.

# Goals
- Improve Vercel support
- Increase app composition capabilities

# Example

``` ts
//index.astro
export const runtime = "edge";
```

**Comment by joe-bell** [comment]
Echoing that this would be wonderful ✨ 

Until https://github.com/lovell/sharp/pull/3522#issuecomment-1519857398 or similar is merged, I'm assuming it's unlikely the "Assets" service will support Edge anytime soon

Being able to opt-in to Edge will help to ensure the app is as fast as possible on dynamic pages that don't depend on Node internals

**Comment by karelnagel** [comment]
Any updates on this?

**Comment by matthewp** [comment]
We dropped support for Vercel Edge as an adapter in `@astrojs/vercel`. Instead edge is used as Edge Middleware which is where it really shines. Closing as we won't be adding this sort of configuration (at least not unless things change otherwise).
